### ISSUE TITLE
Can not use ShouldHaveValidationErrorFor with "collection accessor" string

### ISSUE THREAD

**Issue by sm-g** [JeremySkinner/FluentValidation#1256]
From "Advanced test extensions" at https://fluentvalidation.net/testing I see example:

```
// You can also use a string name for properties that can't be easily represented with a lambda, eg:
result.ShouldHaveValidationErrorFor("Addresses[0].Line1");
```
Such thing does not work in version 8.5.0. Removing `[0]` part from string makes it "working" (i.e. `ShouldHaveValidationErrorFor` detects that there is validation error, but of course cannot distinguish error in [0] from error in [1]). So I have to write something like (NUnit):

```
Assert.IsFalse(result.IsValid);
Assert.That(result.Errors.Select(x => (x.PropertyName, x.ErrorCode)),
         Contains.Item(("Providers.Keys[1]", "NotEmptyValidator")));
```

**Comment by JeremySkinner** [comment]
Thanks, I'll make sure this gets fixed for a future version.

**Comment by Red-F** [comment]
In ValidatorTestExtensions.cs:


```c#
    internal static IEnumerable<ValidationFailure> ShouldHaveValidationError(IList<ValidationFailure> errors, string propertyName) {
      var failures = errors.Where(x => NormalizePropertyName(x.PropertyName) == propertyName
                                       || (string.IsNullOrEmpty(x.PropertyName) && string.IsNullOrEmpty(propertyName))
                                       || propertyName == MatchAnyFailure
                                       ).ToArray();

```
which leads to:
```c#
		private static string NormalizePropertyName(string propertyName) {
			return Regex.Replace(propertyName, @"\[.*\]", string.Empty);
		}

```

**Comment by JeremySkinner** [comment]
Yep, I already have a fix for it, just not merged it yet. https://github.com/JeremySkinner/FluentValidation/pull/1257

**Comment by JeremySkinner** [comment]
Implemented in 8.6
